src/main/java/tasks/Deadline.java

```diff
@@ -0,0 +1,60 @@
+package tasks;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * The class representing a Deadline task.
+ *
+ * @author Gallen Ong
+ */
+public class Deadline extends Task {
+    protected LocalDate by;
```

Comment:
maybe this variable can be named using a noun instead, since "by" can have multiple meanings